resource_booking/models/resource_booking.py

```diff
@@ -0,0 +1,753 @@
+            # other items appear to make sense.
+            if item1_date < item0_date:
+                continue
+            # Intervals that aren't on the running tally date break the streak.
+            # This check is for malformed data in `available_intervals` where a
+            # day is skipped.
+            if item0_date != tally_date or item1_date != tally_date:
+                break
+            # Intervals that aren't on the end date should end at 23:59 (and any
+            # number of seconds).
+            if item1_date != end_date and (item[1].hour != 23 or item[1].minute != 59):
+                break
+            # Intervals that aren't on the start date should start at 00:00 (and
+            # any number of seconds).
+            if item0_date != start_date and (item[0].hour != 0 or item[0].minute != 0):
+                break
+            # The next interval should be on the next day.
+            tally_date += timedelta(days=1)
+        else:
+            return True
+    return False
+
+
+class ResourceBooking(models.Model):
+    _name = "resource.booking"
+    _inherit = ["mail.thread", "mail.activity.mixin", "portal.mixin"]
+    _description = "Resource Booking"
+    _order = "start DESC"
+    _sql_constraints = [
+        (
+            "combination_required_if_event",
+            "CHECK(meeting_id IS NULL OR combination_id IS NOT NULL)",
+            "Missing resource booking combination.",
+        ),
+        (
+            "unique_meeting_id",
+            "UNIQUE(meeting_id)",
+            "Only one event per resource booking can exist.",
+        ),
+    ]
+
+    active = fields.Boolean(default=True)
+    meeting_id = fields.Many2one(
+        comodel_name="calendar.event",
+        string="Meeting",
+        context={"default_res_id": False, "default_res_model": False},
+        copy=False,
+        index=True,
+        ondelete="set null",
+        help="Meeting confirmed for this booking.",
+    )
+    categ_ids = fields.Many2many(string="Tags", comodel_name="calendar.event.type")
+    combination_id = fields.Many2one(
+        comodel_name="resource.booking.combination",
+        string="Resources combination",
+        compute="_compute_combination_id",
+        copy=False,
+        domain="[('type_rel_ids.type_id', 'in', [type_id])]",
+        index=True,
+        readonly=False,
+        states={"scheduled": [("required", True)], "confirmed": [("required", True)]},
+        store=True,
+        tracking=True,
+    )
+    combination_auto_assign = fields.Boolean(
+        string="Auto assigned",
+        default=True,
+        help=(
+            "When checked, resource combinations will be (un)assigned automatically "
+            "based on their availability during the booking dates."
+        ),
+    )
+    name = fields.Char(index=True, help="Leave empty to autogenerate a booking name.")
+    description = fields.Html()
+    partner_id = fields.Many2one(
```

Comment:
This is not correct according v15:

https://github.com/OCA/calendar/blob/15.0/resource_booking/models/resource_booking.py#L126

The compute/inverse is in this field, not in `partner_ids`.

Reply:
`partner_ids` was added to 15.0 after I suggested it for 16.0. Probably I don't need this field after all.

@pedrobaeza Feel free to create a new commit to do the correction. You may remove `partner_ids` if you want.

Reply:
But I think the field may be practcal in some scenarios, even though I don't need it right now.

Reply:
@victoralmau please resume this migration including all the 15.0 stuff. Please complete and include #114 as well.